one-shot-device/src/devices/device-registration.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@nestjs/common';
+import { Model } from 'mongoose';
+import { defaultConfig } from '../configuration/configuration';
+import { InjectModel } from '@nestjs/mongoose';
+import { CreateDeviceRegistrationDto } from './dto/create-device-registration.dto';
+import { DeviceRegistrationDocument, DeviceRegistration } from './schemas/device-registration.schema';
+import { v4 as uuidv4 } from 'uuid';
+import { IdentityClient, ChannelClient } from 'iota-is-sdk';
+
+@Injectable()
+export class DeviceRegistrationService {
+	constructor(
+		@InjectModel(DeviceRegistration.name)
+		private readonly deviceRegistrationModel: Model<DeviceRegistrationDocument>
+	) {}
+
+	async createChannelAndIdentity() {
+		const channelClient = new ChannelClient(defaultConfig);
+		const identityClient = new IdentityClient(defaultConfig);
+
+		// create device identity
+		const deviceIdentity = await identityClient.create('my-device' + Math.ceil(Math.random() * 1000));
+
```

Comment:
please add a check `if(deviceIdentity == null)` and return an error if it is null